devstack/devstack.dockerfile

```diff
@@ -0,0 +1,68 @@
+FROM ubuntu:xenial
+ARG USER_NAME
+ENV USER_NAME ${USER_NAME:-root}
+ARG USER_ID
+ENV USER_ID ${USER_ID:-0}
+
+#
+# Install dependencies (this should all be in debian_packages.lst but not sure yet why it's not already)
+#
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y git sudo python python3-dev python3 python3-venv ca-certificates unzip openssh-server mongodb mysql-server mysql-client unzip gcc make
```

Comment:
Why not start from image `python:3.5` instead of installing it?

Additionally, instead of installing all the components needed on a single container, why not configure standalone `postgres:10.7`, `mongo` and `consul` containers and link them with the Ocim container using `docker-compose`? (similarly to how it's done with the Open edX devstack)

Reply:
It's a matter of convenience and is debatable indeed! My reasons for preferring to start from a base Xenial container instead of python:3.5 is that I know exactly what this image contains. I would have to investigate what python:3.5 has or what it is missing. Is it based on Xenial? etc.

Regarding the services, I have a strong preference for running them in a all-in-one container for development convenience rather than assembling them with docker-compose.yml, because:

* It is what Vagrant does currently
* It is easier to debug, restart, reconfigure a depedency running on the same machine rather than in another container
* Developers would have to learn a different way of interacting with servers compared to what they are currently used to (i.e. all of them running in the same Vagrant VM)

Reply:
@singuliere Regarding your comments:

> Regarding the services, I have a strong preference for running them in a all-in-one container for development convenience rather than assembling them with docker-compose.yml

If this is the main goal, using docker will only make it harder, because of the mount points and permissions, which Vagrant already takes care for us.
The goal of having a docker devstack with multiple containers is to have each service of the system running independently and without interfering with each other's packages and dependencies, and putting everything on the same container makes Docker basically the same as the current Vagrant devstack + tech debt.

> It is easier to debug, restart, reconfigure a dependency running on the same machine rather than in another container

This point is debatable, since the goal of having different containers is to make the services independent and with less risk of updating one component and breaking others.

> Developers would have to learn a different way of interacting with servers compared to what they are currently used to (i.e. all of them running in the same Vagrant VM)

I agree with this, but we already work mainly on Open edX using the docker devstack, so it wouldn't be that much of news.

My point is that it is only worth switching from our Vagrant devstack to a Docker devstack if:
- The docker based devstack improves or eases our current workflow
- Doesn't add a significant amount of tech debt
- Completely replaces our Vagrant devstack

Another point that should be raised is that currently, we use the same playbooks to create the devstack that we use to deploy Ocim (changing some roles, of course), so we also have the plus side of testing our current deployment strategies on the Vagrant machine.

To move this forward, I will have to start an internal discussion with the team to see if switching to a Docker based devstack is in the interest of the project and if we have the budget to get this moving forward.

Thanks for your patience and your contribution! I'll get back to you as soon as I have some news.